src/native/crt.c

```diff
@@ -286,6 +286,18 @@ bool aws_jni_check_and_clear_exception(JNIEnv *env) {
     return exception_pending;
 }
 
+bool aws_jni_get_and_clear_exception(JNIEnv *env, jthrowable *out) {
+    bool exception_pending = (*env)->ExceptionCheck(env);
+    if (exception_pending) {
+        (*env)->DeleteGlobalRef(env, *out);
```

Comment:
It feels unsafe to `DeleteGlobalRef(env, *out);` I don't see where we initialize the `callback_data->java_exception`, does  `DeleteGlobalRef` handle uninitialized object well?

Reply:
Discussed offline, it seems to work since callback_data->java_exception will be NULL for the first time.

Reply:
JNI code is not easy to read, but ultimately I think it calls down to here, and yeah NULL is OK

https://github.com/corretto/corretto-8/blob/790d5c6cdb98dcbd196bf3b1140964aee5cdb9f4/src/hotspot/src/share/vm/runtime/jniHandles.cpp#L129-L134